atc/routes.go

```diff
@@ -161,6 +163,9 @@ var Routes = rata.Routes([]rata.Route{
 	{Path: "/api/v1/teams/:team_name/pipelines/:pipeline_name/resources", Method: "GET", Name: ListResources},
 	{Path: "/api/v1/teams/:team_name/pipelines/:pipeline_name/resource-types", Method: "GET", Name: ListResourceTypes},
 	{Path: "/api/v1/teams/:team_name/pipelines/:pipeline_name/resources/:resource_name", Method: "GET", Name: GetResource},
+	{Path: "/api/v1/teams/:team_name/pipelines/:pipeline_name/resources/:resource_name/cache", Method: "DELETE", Name: ClearResourceCaches},
+	{Path: "/api/v1/teams/:team_name/pipelines/:pipeline_name/resources/:resource_name/cache/:resource_config_version_id", Method: "DELETE", Name: ClearResourceCache},
```

Comment:
I think taking in a `resource_config_version_id` for the ClearResourceCache endpoint might not be worth the trouble because in the database, a resource cache points to a specific version and not a `resource_config_version_id`. When a user runs the fly command for clearing their resource cache, they pass in a version and then the fly command would have to convert that version to a `resource_config_version_id`, call the ClearResourceCache endpoint with it and then the endpoint would have to convert that id back to a version and use that to find the resource cache to delete. 

To that point, it might be easier to just have the `ClearResourceCaches` endpoint be able to also take in a `version` as a param. That way we can straight up use the version passed in from the user to find the resource cache associated to it. In this case, we would also not need to add a `ClearResourceCache` endpoint and just need `ClearResourceCaches`.